locked-asset/factory/src/locked_asset_token_merge.rs

```diff
@@ -107,6 +94,62 @@ pub trait LockedAssetTokenMergeModule:
         Ok((sum_amount, attrs))
     }
 
+    fn calculate_new_unlock_milestones(
+        &self,
+        unlock_epoch_amount_merged: &ArrayVec<
+            EpochAmountPair<Self::Api>,
+            DOUBLE_MAX_MILESTONES_IN_SCHEDULE,
+        >,
+        amount_total: &BigUint,
+    ) -> ManagedVec<UnlockMilestone> {
+        let mut unlock_milestones_merged =
+            ArrayVec::<UnlockMilestone, MAX_MILESTONES_IN_SCHEDULE>::new();
+
+        for el in unlock_epoch_amount_merged.iter() {
+            let unlock_percent = &(&el.amount * PERCENTAGE_TOTAL) / amount_total;
+
+            //Accumulate even the percents of 0
+            unlock_milestones_merged.push(UnlockMilestone {
+                unlock_epoch: el.epoch,
+                unlock_percent: unlock_percent.to_u64().unwrap() as u8,
```

Comment:
Use `ok_or` instead of `unwrap`.

Reply:
added ok_or, but unwrap is still needed. the method does not return a scresult